### PR TITLE
fix: Use build.include sha commit version of theia during release

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Signed-off-by: Mykhailo Kuznietsov <mkuznets@redhat.com>

### What does this PR do?
If build include contains THEIA_COMMIT_SHA value, then use it as theia version, otherwise use next version

### What issues does this PR fix or reference?

closes https://github.com/eclipse/che/issues/20221

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

**Review by azatsarynnyy — APPROVED** [review]

File: make-release.sh [hunk 1]
```diff
@@ -62,9 +62,14 @@ if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then
 fi
 
 apply_files_edits () {
-  THEIA_VERSION=$(curl --silent http://registry.npmjs.org/-/package/@theia/core/dist-tags | sed 's/.*"next":"\(.*\)".*/\1/')
+  BUILD_INCLUDE_THEIA_COMMIT_SHA=$(grep -e "^THEIA_COMMIT_SHA=" build.include | cut -d '=' -f2 | tr -d '"')
+  if [ -z "${BUILD_INCLUDE_THEIA_COMMIT_SHA}"]; then
+    THEIA_VERSION=${BUILD_INCLUDE_THEIA_COMMIT_SHA}
```
Comment (benoitf) [inline make-release.sh:67]: Here it won't work

THEIA_VERSION format is for example
1.16.0-next.3716c071
1.15.0

and not the sha1

so we need to take first digits of BUILD_INCLUDE_THEIA_COMMIT_SHA and then search a package matching this sha1


for example let say commit sha1 is `3716c0716f4f7fe30ef5b36faff4f85e833b20be` we need to take `3716c071`
```bash 
$ npm view @theia/core versions | grep 3716c071                                                                                                                                                                               
'1.16.0-next.3716c071',
```

then we can take `1.16.0-next.3716c071` for THEIA_VERSION

Comment (benoitf) [reply make-release.sh:67]: curl command to get all versions
```bash
$ curl -H "Accept: application/vnd.npm.install-v1+json" https://registry.npmjs.org/@theia/core | jq -r '.versions | keys[]'
...

1.7.0-next.ff9b62f9
1.8.0
1.8.0-next.16fec095
1.8.0-next.1e603511
1.8.0-next.33886b52
1.8.0-next.56c22749
1.8.0-next.5c164f7f
1.8.0-next.61200efc
1.8.0-next.632a67a5
1.8.0-next.cd85908d
1.8.0-next.e2098321
1.8.0-next.fd0a8ce0
1.8.0-next.fea40a2c
1.8.1
...
```

File: make-release.sh [hunk 2]
```diff
@@ -62,9 +62,14 @@ if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then
...
+  else
+    THEIA_VERSION=$(curl --silent http://registry.npmjs.org/-/package/@theia/core/dist-tags | sed 's/.*"next":"\(.*\)".*/\1/')
```
Comment (nickboldt) [inline make-release.sh:69]:
```
curl -sSL http://registry.npmjs.org/-/package/@theia/core/dist-tags | jq -r .next
```

**Review by benoitf — APPROVED** [review]
let's try 👍

**Pull request merged**